src/argus_htmx/themes/utils.py

```diff
@@ -1,21 +1,54 @@
+from importlib.resources import files
+from pathlib import Path
+from re import findall
+
 from django.conf import settings
-from argus_htmx import settings as argus_htmx_settings
+from django.core.exceptions import ImproperlyConfigured
 
+from argus_htmx import settings as fallbacks
 
-def get_themes():
-    return getattr(settings, "DAISYUI_THEMES", argus_htmx_settings.DAISYUI_THEMES)
 
+__all__ = [
+    "get_theme_names",
+    "get_theme_default",
+]
 
-def get_theme_names():
-    themes = get_themes()
+
+def get_themes_from_setting():
+    themes_setting = getattr(settings, "DAISYUI_THEMES", fallbacks.DAISYUI_THEMES)
     theme_names = []
-    for theme in themes:
+    for theme in themes_setting:
         if isinstance(theme, str):
             theme_names.append(theme)
         elif isinstance(theme, dict):
             theme_names.extend(theme.keys())
     return theme_names
 
 
+def get_stylesheet_path():
+    return getattr(settings, "STYLESHEET_PATH", fallbacks.STYLESHEET_PATH)
+
+
+def get_themes_from_css():
+    THEME_NAME_RE = "(?P<theme>\w+)"
+    DATA_THEME_RE = f"\[data-theme={THEME_NAME_RE}\]"
+
+    static_url = Path(settings.STATIC_URL).relative_to("/")
+    stylesheet_path = static_url / get_stylesheet_path()
+    styles_css = files("argus_htmx").joinpath(stylesheet_path).read_text()
+
+    return findall(DATA_THEME_RE, styles_css)
+
+
+def get_theme_names():
+    themes_from_setting = set(get_themes_from_setting())
+    themes_from_css = set(get_themes_from_css())
+    installed_themes = themes_from_setting | themes_from_css
+    all_themes = themes_from_setting & themes_from_css
+    if all_themes != installed_themes:
+        raise ImproperlyConfigured("Themes in settings is out of sync with themes installed")
+    return installed_themes
```

Comment:
nitpick: I think there is a mixup between the names installed_themes and all_themes. `|` is the set union, which includes all items in either set, while `&` is the set intersection which only includes items that are in both sets. the union is a superset of the intersection. It makes more sense that `all_themes` is the superset and `installed_themes` the subset.

In this case there this has no effect on the functionality though


you could also use `^` to get the symmetric difference of both sets. If it's not empty, it's ImproperlyConfigured

```suggestion
    if themes_from_setting ^ themes_from_css:
        raise ImproperlyConfigured("Themes in settings is out of sync with themes installed")
    return themes_from_setting 
```

Reply:
Fixed in later commits